src/freenet/clients/http/ToadletContextImpl.java

```diff
@@ -437,6 +437,8 @@ static void sendReplyHeaders(
 		}
 		String contentSecurityPolicy = generateCSP(allowScripts, allowFrames);
 		mvt.put("content-security-policy", contentSecurityPolicy);
+		mvt.put("cross-origin-embedder-policy", "require-corp");
+		mvt.put("cross-origin-opener-policy", "same-origin");
 		mvt.put("x-content-security-policy", contentSecurityPolicy);
```

Comment:
This seems very reasonable (although the headers are new to me). But could you move the new stuff down a few lines so it is not smack in the middle of several lines related to CSP? :smile:

Reply:
I don't like adding something headers after the x-headers, so I moved them before CSP.